js_runner.go

```diff
@@ -66,13 +68,39 @@ func (runner *JSRunner) Init(funcSource string) error {
 		return otto.UndefinedValue()
 	})
 
+	runner.js.Set("console", map[string]interface{}{
+		"error": func(s string) { errorMessageCallback(s)},
+		"log": func(s string) { logMessageCallback(s)},
+	})
+
 	if _, err := runner.SetFunction(funcSource); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+var logMessageCallback = defaultLogFunction
+var errorMessageCallback = defaultLogFunction
+
+var logCallbackLock sync.Mutex
+
+func (runner *JSRunner) SetConsoleErrorCallback(logFn func(string)){
+	logCallbackLock.Lock()
+	errorMessageCallback = logFn
+	logCallbackLock.Unlock()
+}
+
+func (runner *JSRunner) SetConsoleLogCallback(logFn func(string)){
+	logCallbackLock.Lock()
+	logMessageCallback = logFn
+	logCallbackLock.Unlock()
+}
+
+func defaultLogFunction(s string) {
+	fmt.Fprintln(os.Stdout, s)
```

Comment:
Is there a reason to not just use fmt.Println here?

Reply:
I'm not entirely sure. In all honesty I copied otto's default logger as closely as possible and it used `fmt.Fprintln`. I suppose it may be required to ensure sg-bucket works properly when not using SG?